agora/shopper_insights_api/src/app.py

```diff
@@ -68,7 +71,7 @@ def video_feed():
         return "Invalid JSON format", 400
     
     camera_name = data["cameraName"]       
-    return Response(generate(data, camera_name), mimetype='multipart/x-mixed-replace; boundary=frame')
+    return Response(executor.submit(generate, data, camera_name).result(), mimetype='multipart/x-mixed-replace; boundary=frame')
```

Comment:
If i understand this correctly, every time someone calls the video_feed, you create a new thread, even if it's the same camera. But then the camera processor and captures are the always the same 2 threads for a specific camera - is that correct?

Reply:
agreed, we dont need to pool the api calls themselves, just the video processor. copilot gonna copilot